torch/backends/cudnn/__init__.py

```diff
@@ -81,6 +83,29 @@ def is_acceptable(tensor):
 CUDNN_RNN_ALGO_PERSIST_DYNAMIC = 2
 
 
+@contextmanager
+def flags(enabled=False, benchmark=False, deterministic=False, verbose=False):
+    # capture the original values
+    orig_enable = enabled
+    orig_benchmark = benchmark
+    orig_deterministic = deterministic
+    orig_verbose = verbose
+
+    # set the new values
+    enabled = enabled
+    benchmark = benchmark
+    deterministic = deterministic
+    verbose = verbose
```

Comment:
um this doesn't work. The whole context manager is a no-op right now. You're just assigning local variable to the same variable, as the global flags got shadowed. You might need to fiddle with `globals()`, or use a helper function that will take arguments with different names (i.e. make `flags` call `_flags(_enabled=enabled, _benchmark=benchmark, ...)`

Reply:
yes, n00b errors, my bad! thanks for catching!